### DISCUSSION TITLE
FirebaseAuthentication.signInWithEmailAndPassword throws internal error when email not found

### DISCUSSION THREAD

**Discussion by dom-the-dev** [capawesome-team/capacitor-firebase#609] [Q&A] [answered]
Hey folks,

I've been using **FirebaseAuthentication.signInWithEmailAndPassword** for login functionality in my app, and it's been working smoothly for the most part. 
However, I've recently encountered an issue where, when users enter an incorrect email or password, instead of receiving specific error codes like "auth/user-not-found" or "auth/wrong-password," I'm getting a generic "An internal error has occurred" message.

The login functionality still works perfectly fine with correct credentials. It's only when users input incorrect information that I encounter this issue.

Has anyone else experienced a similar issue recently? Could this be a result of changes to the Firebase Authentication API or a bug on my end? Any insights or suggestions would be greatly appreciated.

Thanks for your help!

**Comment by robingenz** [comment]
I've just tested it and you should get a `auth/invalid-credential` error which means that either the username or the password is incorrect. As far as I know, `auth/user-not-found` and `auth/wrong-password` are no longer returned in order to increase security.

![grafik](https://github.com/capawesome-team/capacitor-firebase/assets/13857929/2032852c-6cdd-4e99-928b-511c2c88ff0c)

**Comment by dom-the-dev** [comment]
Hey Robin, appreciate your response.

When I test this in the browser, I indeed get the same error code. However, when I test this in the Simulator, I get error code `internal-error`

web:
<img width="716" alt="Bildschirmfoto 2024-04-20 um 09 04 16" src="https://github.com/capawesome-team/capacitor-firebase/assets/26629137/4aba73a4-f205-45b8-8360-3fedbf84a6ef">

simulator:
<img width="890" alt="Bildschirmfoto 2024-04-20 um 09 00 42" src="https://github.com/capawesome-team/capacitor-firebase/assets/26629137/f54c0fa7-9aba-49f2-9418-d4d25c717788">

**Reply by robingenz** [reply, marked as answer]
Unfortunately, I cannot reproduce this. My screenshot above is from the simulator. Make sure to use the latest plugin version.

**Reply by dom-the-dev** [reply]
I just upgraded to Capacitor 6. Now I am getting the same error message as you. Thank you for your help.
